debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotMySQLUtility.java

```diff
@@ -0,0 +1,600 @@
+        for(String database : databases) {
+            schema.applyDdl(source,
+                            database,
+                            "DROP DATABASE IF EXISTS " + quote(database),
+                            statementConsumer);
+        }
+    }
+
+    public void createDropTableEvents(Collection<TableId> tables,
+                                      DatabaseStatementStringConsumer statementConsumer) throws SQLException {
+        MySqlSchema schema = context.dbSchema();
+        SourceInfo source = context.source();
+        for (TableId tableId : tables) {
+            schema.applyDdl(source,
+                            tableId.schema(),
+                            "DROP TABLE IF EXISTS " + quote(tableId),
+                            statementConsumer);
+        }
+    }
+
+    public void createCreateDatabaseEvents(Collection<String> databases,
+                                           DatabaseStatementStringConsumer statementConsumer) throws SQLException {
+        MySqlSchema schema = context.dbSchema();
+        SourceInfo source = context.source();
+        for (String database: databases) {
+            schema.applyDdl(source, database, "CREATE DATABASE " + quote(database), statementConsumer);
+        }
+    }
+
+    /**
+     * Create 'CREATE' events for the given tables.
+     * @param tables The collection of tables to create 'CREATE' events for.
+     */
+    public void createCreateTableEvents(Collection<TableId> tables,
+                                        DatabaseStatementStringConsumer statementConsumer) throws SQLException {
+        // todo I could make this a bit less repetitive by grouping the tables by database.
+        MySqlSchema schema = context.dbSchema();
+        SourceInfo source = context.source();
+        for (TableId tableId : tables) {
+            String database = tableId.catalog();
+            // put us in the correct database
+            // todo is this really needed?
+            schema.applyDdl(source, database, "USE " + quote(database), statementConsumer);
+            // create the table.
+            sql.set("SHOW CREATE TABLE " + quote(tableId));
+            mysql.query(sql.get(), rs -> {
+                if (rs.next()) {
+                    schema.applyDdl(source,
+                                    database,
+                                    rs.getString(2),
+                                    statementConsumer);
+                }
+            });
+        }
+    }
+
+    /**
+     *
+     * @param recorder
+     * @param bufferedRecordQueue
+     * @throws SQLException
+     */
+    // todo method input order?
+    // todo buffer a given list of tables (those tables must be locked, of course)
+    // but that way we can lock all the tables but only buffer/read some tables.  Which is useful because then we
+    // can release the lock earlier than we could normally.
+    public void bufferLockedTables(List<TableId> tableIdsToBuffer,
```

Comment:
A description of this method would be good. Also I think it still is very long, can we extract further things into separate methods?